aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,190 @@
+
+To help developers get started deploying with .NET native AOT in ASP.NET Core, use the:
+
+* AOT API template which includes customizations to remove unsupported components from the app.
+* ```dotnet new``` command to create a new ASP.NET Core API app that is configured to work with .NET native AOT:
+
+```powershell
+PS> dotnet new api -aot -n MyFirstAotWebApi && cd MyFirstAotWebApi
+The template "ASP.NET Core API" was created successfully.
+
+Processing post-creation actions...
+Restoring C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj:
+  Determining projects to restore...
+  Restored C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj (in 302 ms).
+Restore succeeded.
+```
+
+Before we look more closely at the code in the template, let's make sure that it can but published using .NET native AOT correctly by using the following command (note the versions of .NET 8.0+ that you are using may vary from the output shown below):
+
+```
+PS> dotnet publish
+MSBuild version 17.<version> for .NET
+  Determining projects to restore...
+  Restored C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj (in 241 ms).
+C:\Code\dotnet\aspnetcore\.dotnet\sdk\8.0.<version>\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIde
+ntifierInference.targets(287,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotne
+t-support-policy [C:\Code\Demos\MyFirstAotWebApi\MyFirstAotWebApi.csproj]
+  MyFirstAotWebApi -> C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\MyFirstAotWebApi.dll
+  Generating native code
+  MyFirstAotWebApi -> C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\publish\
+```
+
+Note: The preceding output my differ from what you see depending on the version of  .NET 8 used.
+Review the contents of output directory:
+
+```
+PS> dir bin\Release\net8.0\win-x64\publish
+
+    Directory: C:\Code\Demos\MyFirstAotWebApi\bin\Release\net8.0\win-x64\publish
+
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a---          30/03/2023  1:41 PM        9480704 MyFirstAotWebApi.exe
+-a---          30/03/2023  1:41 PM       43044864 MyFirstAotWebApi.pdb
+```
+
+The API template with AOT enabled produces a binary of about 9.4MB on Windows, although the size may vary depending on the build of .NET 8.0 used. The executable can be run to a machine without the .NET Core runtime installed.
+
+```
+PS> .\bin\Release\net8.0\win-x64\publish\MyFirstAotWebApi.exe
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://localhost:5000
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Production
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: C:\Code\Demos\MyFirstAotWebApi
+```
+
+The `Program.cs` source file contains some changes for publishing to .NET native AOT:
+
+```csharp
```

Comment:
It might be helpful to pull out the parts that are different into separate code blocks to make it more clear to the reader.